### PR TITLE
Support for attachments, screenshots and new SDK for Android

### PR DESCRIPTION
**Pull request by avargaskun**
The following changes are included with this PR:

### Upgrade to HockeyApp SDK 3.7.0 for Android

This is required to include attachments along with feedback reports, which is used by the new API introduced as well by this PR.

The main takeaways of this upgrade are:

* Removed the HockeyApp documentation folder from the plug-in as it is now back in sync with the latest released version and documentation is available online (also, it was giving me some merge headaches)
* The SDK moved from a JAR to an AAR file, and as such I had to introduce a gradle file to link the AAR (otherwise the package was not being properly recognized)

#### Add new API: `feedbackModal`

This new API allows callers to include either or both of the following:

* A screen shot taken at the moment the API is called. Included in the feedback report as a JPEG file.
* An object serialized as JSON. Included in the feedback report as a text file.

These methods are a really convenient of including extra information in the report without having to force a crash. Applications that allow users to submit feedback based on weird/buggy behavior that is entirely recoverable will not need to crash and restart. I'm using this new API in a shake-to-send bug report feature in my app.

The main takeaways of this API are:

* I called it `feedbackModal` because in iOS it really has different behavior, showing a modal window as opposed to the **Feedback Manager** window that shows up when calling just `feedback`. Android does not have have the **Feedback Manager** view and in turn both API call the same underlying SDK function.
* When using a CrossWalk View the screen shot has to be taken using internal XWalk Engine functionality which was just recently exposed in version 18 (beta), as well as the most recent (not yet released) version of the [cordova-crosswalk-engine](https://github.com/fujunwei/cordova-crosswalk-engine). 
	* Fortunately, the changes do not add a build-time dependency on any of the XWalk classes/plugins however, if the new screen shot functionality is not present in the CrossWalk View or plugin versions used along with this plugin, the screen shot will come out blank.
	* There should be no changes required to this plugin once both the new engine and plugin for CrossWalk are officially released.

#### Other notes

README.md has been updated to reflect the changes introduced in this PR.

I've tested the new function in iOS 9 and Android 5 physical devices, built with Cordova 6.0.0

### CODE REVIEW

**Comment by avargaskun** [comment]
Hey @peutetre I'm submitting this PR for your consideration. I think the new functionality is worth sharing. Let me know what you think!

File: README.md [hunk 1]
```diff
@@ -53,3 +62,12 @@ Can be used to test the crash reporting feature of HockeyApp.
 hockeyapp.addMetaData(success:function, error:function, data:object):void
 ```
 Will be displayed in crash reports in HockeyApp.
+
+## CrossWalk considerations
+
+An application that is hosted in a CrossWalk View will show a blank screenshot unless the following requirements are met:
```
Comment (piotrowski) [inline README.md]: This sentence could be clearer, that it refers to the screenshots created by `hockeyapp.feedbackModal`.

**Comment by piotrowski** [comment]
Seems like some nice functionality!

**Comment by avargaskun** [comment]
Thanks for the comments @piotrowski - I've updated the readme file to reflect the suggested change.

**Comment by lostintangent** [comment]
It would be awesome to see this merged!

**Comment by lostintangent** [comment]
@avargaskun We just updated the Android SDK to use the 4.1.0 release. Could you update your PR to exclude the SDK update, since it would no longer be necessary to make this fix? Thanks!b

**Comment by lostintangent** [comment]
I'm closing this PR since it can't be merged into master due to conflicts. Could you update your branch and re-submit your PR whenever you get a chance? Thanks so much!

**Pull request closed without merging**